test/unit/builtins/numbers.spec.ts

```diff
@@ -69,6 +69,45 @@ test.each([
     util.testExpressionTemplate`(${value}).toString(2)`.expectToEqual(luaNativeSpecialNumString);
 });
 
+// frac=100 fails as result can't fit
+const toFixedFractions = [undefined, 0, 1, 1.23456, 2, Math.PI, 5, 10, 80];
+// val=x.5 fails as rounding differ
+const toFixedValues = [
+    -1,
+    0,
+    1,
+    1.2345678,
+    -20391,
+    3.1415,
+    Math.PI,
+    2.7182,
+    -34910.3,
+    1.45,
+    1.55,
+    1.444555,
+    1024,
+    1.2,
+    1e19,
+    1e22,
+];
+const toFixedPairs = toFixedValues.flatMap(value => toFixedFractions.map(frac => [value, frac] as const));
```

Comment:
9 * 16 seems a bit excessive, lots of these tests seem to be slight duplicates of each other, i,e if you just have 1.444555 and 1.2 in here you can probably remove the other fractionals

Reply:
1.444555 I was trying to get another bad rounding error, but it seems only `N+0.5` can reliably cause those